cost/aws/rds_eos_instances/aws_eos_rds_instances.pt

```diff
@@ -0,0 +1,807 @@
+    }
+
+    function formatCurrency(value, symbol, separator) {
+      var parts = value.toFixed(2).split(".");
+      var integerPart = parts[0];
+      var decimalPart = parts[1];
+      var formattedInt = "";
+      var count = 0;
+
+      for (var i = integerPart.length - 1; i >= 0; i--, count++) {
+        formattedInt = integerPart[i] + formattedInt;
+        if ((count + 1) % 3 === 0 && i !== 0 && separator !== "") {
+          formattedInt = separator + formattedInt;
+        }
+      }
+      return symbol + formattedInt + "." + decimalPart;
+    }
+    
+    function stripCurrencySymbol(formattedValue, symbol) {
+      if (formattedValue.indexOf(symbol) === 0) {
+        return formattedValue.substring(symbol.length);
+      }
+      return formattedValue;
+    }
+
+    function cleanCurrencyValue(formattedValue, symbol, separator) {
+      var value = formattedValue;
+
+      // Remove symbol if present
+      if (value.indexOf(symbol) === 0) {
+        value = value.substring(symbol.length);
+      }
+
+      // Remove separator (e.g., comma)
+      if (separator !== "") {
+        value = value.split(separator).join("");
+      }
+      return value;
+    }
+
+    // Build EOS lookup
+    var eos_lookup = {};
+    for (var i = 0; i < ds_aws_db_instance_map.length; i++) {
+      var item = ds_aws_db_instance_map[i];
+      var engine = item["engine"] ? item["engine"].toLowerCase() : null;
+      var version = item["engineVersion"];
+      var eos_date = item["endOfSupportDate"];
+
+      if (engine && version && eos_date) {
+        if (!eos_lookup[engine]) {
+          eos_lookup[engine] = {};
+        }
+        eos_lookup[engine][version] = eos_date;
+      }
+    }
+
+    // Build tag map
+    var tag_map = {};
+    _.each(ds_resource_tags, function(tag) {
+      var id = tag.instanceId;
+
+      if (!tag_map[id]) {
+        tag_map[id] = [];
+      }
+   
+      if (!_.isEmpty(tag.key) && !_.isEmpty(tag.value)) {
+        tag_map[id].push(tag.key + "=" + tag.value);
+      }
+
+    });
+
+    function isExcluded(instanceId) {
+      var instanceTags = tag_map[instanceId] || [];
+      if (param_exclusion_tags.length === 0) return false;
+
+      if (param_exclusion_tags_boolean === "Any") {
+        for (var i = 0; i < param_exclusion_tags.length; i++) {
+          if (instanceTags.indexOf(param_exclusion_tags[i]) !== -1) {
+            return true;
+          }
+        }
+        return false;
+      } else {
+        for (var i = 0; i < param_exclusion_tags.length; i++) {
+          if (instanceTags.indexOf(param_exclusion_tags[i]) === -1) {
+            return false;
+          }
+        }
+        return true;
+      }
+    }
+
+    // Build billing center lookup
+    var billing_center_lookup = {};
+    _.each(ds_billing_centers, function(bc) {
+      if (bc.id && bc.name) {
+        billing_center_lookup[bc.id] = bc.name;
+      }
+    });
+
+    function getBillingCenter(instanceId) {

```

Comment:
In most cases, functions should all be at the top of the script block rather than mixed in with other code.